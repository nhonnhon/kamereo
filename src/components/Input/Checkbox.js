import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './Checkbox.scss'

class Checkbox extends Component {
  static propTypes = {
    checked: PropTypes.bool,
    blocked: PropTypes.bool,
    children: PropTypes.node,
    text: PropTypes.string,
    center: PropTypes.bool,
    className: PropTypes.string,
    id: PropTypes.string,
    color: PropTypes.string,
    onCheck: PropTypes.func,
  }

  static defaultProps = {
    color: '',
    blocked: false,
  }

  toggle = e => {
    e.stopPropagation()
    this.props.onCheck()
  }

  render() {
    const { children, text, center, className, id, color, blocked, checked } = this.props
    const statusHold = checked ? 'checked' : ''
    const status = blocked ? 'blocked' : statusHold
    return (
      <div
        className={`checkbox ${status} ${className || ''}`}
        style={{ alignItems: center ? 'center' : '' }}
        onClick={this.toggle}
      >
        <div className="icon" />
        <input className="hide" id={id} type="checkbox" />
        {text ? (
          <label className={`pad-left ${color}`} htmlFor={id}>
            {text}
          </label>
        ) : (
          <label className={`pad-left ${color}`} htmlFor={id}>
            <div className={`content`}>{children}</div>
          </label>
        )}
      </div>
    )
  }
}

export default Checkbox
