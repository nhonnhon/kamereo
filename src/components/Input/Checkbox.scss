@import '../../assets/css/base.scss';

.checkbox {
  display: block;
  min-height: 20px;
  position: relative;
  cursor: pointer;
  > .icon {
    height: 20px;
    position: absolute;
    width: 20px;
    border: 1px solid $gray1;
    background-color: $white;
  }
  &.checked {
    > .icon {
      background-color: $green;
    }
  }
  > label {
    display: inline-block;
    font-weight: 300;
    line-height: 1.57em;
    &.white {
      color: #ffffff;
      span,
      p {
        color: #ffffff;
      }
    }
  }
}
