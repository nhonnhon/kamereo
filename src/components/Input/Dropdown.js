import { map, isFunction, find } from 'lodash'
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Text } from 'components'
import { Scrollbars } from 'react-custom-scrollbars'
import './Dropdown.scss'

class Dropdown extends Component {
  static propTypes = {
    icon: PropTypes.string,
    label: PropTypes.string,
    placeholder: PropTypes.string,
    className: PropTypes.string,
    renderOption: PropTypes.func,
    onChange: PropTypes.func,
    options: PropTypes.array.isRequired,
    width: PropTypes.string, //in row
    value: PropTypes.any,
    customStyle: PropTypes.object,
  }

  static defaultProps = {
    // placeholder: 'Please select',
    renderOption: null,
    width: 'auto', //in row
    onChange: () => {},
  }

  constructor(props) {
    super(props)
    this.state = {
      showOptions: false,
      selectedOption: { value: '', label: '' },
      selectedItem: this.props.placeholder,
    }

    this.dropdownRef = React.createRef()
  }

  componentDidMount() {
    const nextOption = find(this.props.options, { value: this.props.value })
    if (nextOption) {
      this.setState({ selectedOption: nextOption })
    }

    const el = this.dropdownRef.current

    el.addEventListener('focusout', () => {
      this.setState({
        showOptions: false,
      })
    })
  }

  componentDidUpdate() {
    if (this.props.value && this.props.value !== this.state.selectedOption.value) {
      const nextOption = find(this.props.options, { value: this.props.value })
      if (nextOption) {
        this.setState({ selectedOption: nextOption })
      } else {
        this.setState({ selectedOption: { value: '', label: '' } })
      }
    }
  }

  get value() {
    return this.state.selectedOption.value
  }

  toggle = () => {
    this.setState(prevState => ({ showOptions: !prevState.showOptions }))
  }

  selectItem = selectedOption => {
    this.setState({ selectedOption })
    this.toggle()
    if (this.props.onChange) this.props.onChange(selectedOption)
  }

  renderOption = option => {
    return (
      <div className="dropdown-option">
        <Text>{option.label}</Text>
      </div>
    )
  }

  render() {
    const { showOptions, selectedOption } = this.state
    const { label, placeholder, className, icon, options, renderOption, width, customStyle } = this.props
    const renderOptionFunc = isFunction(renderOption) ? renderOption : this.renderOption
    const classNames = `dropdown ${showOptions ? 'is-open' : ''}`

    return (
      <div style={{ width }} className={classNames} tabIndex="-1" ref={this.dropdownRef}>
        <Text className="small label">{label}</Text>
        <div onClick={this.toggle} style={customStyle} className={`dropdown-header ${className || ''}`}>
          <div className="row align-left">
            {icon && <img className="mar-right-sm extra-icon" src={icon} alt="" />}
            <div>
              <Text className={`medium  ${selectedOption.value ? 'text-dark' : 'text-placeholder'}`}>
                {selectedOption.value ? selectedOption.label : placeholder}
              </Text>
            </div>
          </div>
          <img className="arrow" src={require('assets/images/icons/collapsible-arrow.svg')} alt="" />
        </div>
        <div className="dropdown-content">
          <Scrollbars autoHeight>
            {map(options, option => {
              return (
                <div
                  onClick={() => {
                    this.selectItem(option)
                  }}
                  key={option.value}
                >
                  {renderOptionFunc(option)}
                </div>
              )
            })}
          </Scrollbars>
        </div>
      </div>
    )
  }
}

export default Dropdown
