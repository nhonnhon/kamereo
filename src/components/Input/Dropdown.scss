@import '../../assets/css/base.scss';

.dropdown {
  outline: none !important;
  position: relative;
  .label {
    padding-bottom: 10px;
  }
  .dropdown-header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    border-radius: 3px;
    overflow: hidden;
    padding: 14px 22px 12px 15px;
    background-color: #ffffff;
    border: 1px solid $gray1;
    .extra-icon {
      width: 32px;
    }
    .arrow {
      transform: rotate(180deg);
      transition: all 0.4s ease-in;
      height: 9px;
    }

    &.is-open {
      .arrow {
        transform: rotate(0);
      }
    }

    &.in-group {
      border-radius: 0;
      border-bottom: 1px solid $gray;
    }
    &.in-group.first-child-in-group {
      border-top-left-radius: 3px;
      border-top-right-radius: 3px;
    }
    &.in-group.last-child-in-group {
      border-bottom-left-radius: 3px;
      border-bottom-right-radius: 3px;
    }

    &.in-row {
      border-radius: 0;
      border-left: 1px solid $gray;
      border-bottom: 1px solid $gray;
      box-shadow: none;
    }
    &.in-row.first-child-in-row {
      border-left: none;
    }
  }
  .dropdown-content {
    position: absolute;
    background: #fafafa;
    z-index: 1;
    transition: 0.3s all ease-in;
    top: 100%;
    width: 100%;
    border-bottom-left-radius: 3px;
    border-bottom-right-radius: 3px;
    max-height: 0;
    overflow: hidden;
    border: 1px solid #eee;
    border-top: none;
    opacity: 0;
    // background-color: #f2f2f2;
    background-color: #ffffff;
  }
  .dropdown-option {
    padding: 10px;
    &:hover {
      // background: #ffffff;
      background: #f2f2f2;
      cursor: pointer;
    }
  }
  &.is-open {
    .dropdown-content {
      max-height: 300px;
      opacity: 1;
    }
  }
}
