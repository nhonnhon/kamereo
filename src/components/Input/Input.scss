@import '../../assets/css/base.scss';

.sc-input-container {
  .extra-icon {
    width: 32px;
  }
  .label {
    padding-bottom: 10px;
    color: $gray;
  }
  .input-container {
    border: 1px solid $gray1;
    padding: 14px 22px 12px 15px;
    border-radius: 3px;
    background: #ffffff;
    display: flex;
    flex: 1;
    flex-direction: column;
    &.focus {
      border: 1px solid $gray;
    }
    input {
      background: #ffffff;
      border: none;
      font-size: 1em;
      font-weight: 400;
      padding: 0;
      &:focus {
        outline-offset: 0;
        outline: none;
      }
      &[disabled] {
        &::placeholder {
          /* Chrome, Firefox, Opera, Safari 10.1+ */
          color: $gray1;
        }
      }

      &::-webkit-input-placeholder {
        /* Chrome/Opera/Safari */
        opacity: 0.4;
      }
      &::-moz-placeholder {
        /* Firefox 19+ */
        opacity: 0.4;
      }
      &:-ms-input-placeholder {
        /* IE 10+ */
        opacity: 0.4;
      }
      &:-moz-placeholder {
        /* Firefox 18- */
        opacity: 0.4;
      }
    }
    textarea {
      background: #ffffff;
      border: none;
      width: 100%;
      resize: none;
      padding: 0;
      &:focus {
        outline-offset: 0;
        outline: none;
      }
      &[disabled] {
        &::placeholder {
          /* Chrome, Firefox, Opera, Safari 10.1+ */
          color: $gray1;
        }
      }
    }
  }
}
