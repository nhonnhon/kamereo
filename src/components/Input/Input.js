import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './Input.scss'

class Input extends Component {
  static propTypes = {
    type: PropTypes.string,
    icon: PropTypes.string,
    label: PropTypes.string,
    readonly: PropTypes.bool,
    disabled: PropTypes.bool,
    password: PropTypes.bool,
    placeholder: PropTypes.string,
    className: PropTypes.string,
    customStyle: PropTypes.object,
    onChange: PropTypes.func,
    value: PropTypes.string,
    onBlur: PropTypes.func,
  }

  static defaultProps = {
    type: 'text',
    readonly: false,
    onChange: () => {},
    onBlur: () => {},
    value: '',
  }

  state = {
    stateValue: this.props.value,
    focus: false,
    type: this.props.password ? 'password' : 'text',
  }

  clear = () => {
    this.setState({ stateValue: '' })
  }

  focus = () => {
    this.setState({ focus: true })
  }

  handleChange = e => {
    const { id, onChange, onChangeId } = this.props
    if (!this.readonly) {
      const value = e.target.value

      onChange && onChange(value)
      onChangeId && onChangeId(id, value)
      this.setState({ stateValue: value })
    }
  }

  handleFocus = () => {
    this.setState({ focus: true })
  }

  handleBlur = () => {
    this.setState({ focus: false })
  }

  handleKeyPress = e => {
    if (e.key === 'Enter') {
      this.props.onEnter && this.props.onEnter()
    }
  }

  render() {
    const { stateValue, type, focus } = this.state
    const { id, label, icon, disabled, className, placeholder, value } = this.props

    return (
      <div className={`sc-input-container ${className || ''} ${label ? '' : 'no-label'}`}>
        {label && <div className={`label text small ${disabled ? 'disabled' : ''}`}>{label}</div>}
        <div className="d-flex align-left flex1">
          {icon && <img className="mar-right-sm extra-icon" src={icon} alt="" />}
          <div className={`input-container ${focus ? 'focus' : ''}`}>
            <input
              key={id}
              autoFocus={this.props.autoFocus}
              value={value || stateValue}
              disabled={disabled}
              type={type}
              placeholder={placeholder}
              onBlur={this.handleBlur}
              onFocus={this.handleFocus}
              onChange={this.handleChange}
              onKeyPress={this.handleKeyPress}
              ref={node => (this.input = node)}
            />
          </div>
        </div>
      </div>
    )
  }
}

export default Input
