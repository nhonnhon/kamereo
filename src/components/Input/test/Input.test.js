import React from 'react'
import { create } from 'react-test-renderer'
import Input from '../Input'

describe('Button component', () => {
  test('Matches the snapshot', () => {
    const button = create(<Input />)
    expect(button.toJSON()).toMatchSnapshot()
  })
})
