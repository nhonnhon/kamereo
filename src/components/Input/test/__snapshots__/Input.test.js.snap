// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<div
  className="sc-input-container  no-label"
>
  <div
    className="d-flex align-left flex1"
  >
    <div
      className="input-container "
    >
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
